### ISSUE TITLE
Ignore formatting revisions in `git blame`

### ISSUE THREAD

**Issue by TechnoPorg** [godotengine/godot-proposals#3834]
### Describe the project you are working on

Contributing to, and in the process figuring out, the Godot Engine.

### Describe the problem or limitation you are having in your project

When trying to understand code, it can give a lot of helpful context to use `git blame`, to see what the code used to be, what files are related to a given change, examples of how the code is used, etc.
A lot of the time, though, instead of the last change being to the code functionality, it will be from an uber-commit that applies a large general change to the entire codebase (I'm looking at you, https://github.com/godotengine/godot/commit/5dbf1809c6e3e905b94b8764e99491e608122261). That's not particularly helpful!

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Git allows you to set a configuration file with a list of revisions to exclude from git blame. I suggest that Godot takes advantage of this feature. Some candidates for commits to exclude that I've found are:
https://github.com/godotengine/godot/commit/5dbf1809c6e3e905b94b8764e99491e608122261
https://github.com/godotengine/godot/commit/0be6d925dc3c6413bce7a3ccb49631b8e4a6e67a
https://github.com/godotengine/godot/commit/e956e80c1fa1cc8aefcb1533e5acf5cf3c8ffdd9

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The `blame.IgnoreRevsFile` git configuration option can be set locally to point to a file containing a list of revisions to exclude from `git blame`.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This is a meta enhancement, which would only affect engine development, not engine usage.

### Is there a reason why this should be core and not an add-on in the asset library?

See above.

**Comment by Calinou** [comment]
Does `blame.IgnoreRevsFile` have a default location? If not, how should we supply a path that can be committed to the repository and used by Git automatically?

**Comment by bluenote10** [comment]
I noticed this issue as well, and my experience with the proposed technique in other projects was quite well.

> Does blame.IgnoreRevsFile have a default location within the repository (akin to .gitattributes)? If not, how should we supply a file that can be committed to the repository and used by Git automatically?

To my knowledge there is unfortunately no way to automatically apply it. What works reasonable well in practice is to use a similarly named top-level file like `.gitignorecommits` and include the documentation inline with the ignored commits:

```sh
# -----------------------------------------------------------------------------
# This file contains a list of git commits that should be hidden from the
# regular git history. Typically this includes commits involving mass auto
# formatting or other normalizations. To apply the ignore list in your local
# git client you must run:
#
#   git config blame.ignoreRevsFile .git-blame-ignore-revs .gitignorecommits
#
# -----------------------------------------------------------------------------

# clang format related commits:
5dbf180
0be6d92
e956e80
```

People who see the file and look into it immediately understand the idea and have a chance to opt-in into the ignore feature if they like, or chose to not do it if they don't. Perhaps enforcing the ignore list automatically wouldn't make everyone happy anyway.

**Comment by Calinou** [comment]
> To my knowledge there is unfortunately no way to automatically apply it.

Too bad… this also means github.com will not obey this file.

**Comment by bluenote10** [comment]
> Too bad… This also means github.com will not obey this file.

Yes indeed, as of now, I don't think they support it. But it is a [highly demanded feature](https://github.community/t/support-ignore-revs-file-in-githubs-blame-view/3256), and it is not unlikely that code hosting sites like GitHub support it in the long term.

**Comment by TechnoPorg** [comment]
GitHub has confirmed that they are working on implementing this feature for their blame view: https://github.com/github/feedback/discussions/5033#discussioncomment-2237184

**Comment by Calinou** [comment]
GitHub now implemented support for `.git-blame-ignore-revs`: https://github.blog/changelog/2022-03-24-ignore-commits-in-the-blame-view-beta/
